SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java

```diff
@@ -998,49 +996,27 @@ public void visitSort(LogicalSort sort) {
         this.circuit.addOperator(agg);
 
         // Generate comparison function for sorting the vector
-        DBSPExpression comparators = null;
+        DBSPComparatorExpression comparator = new DBSPNoComparatorExpression(sort, inputRowType);
         for (RelFieldCollation collation: sort.getCollation().getFieldCollations()) {
             int field = collation.getFieldIndex();
             RelFieldCollation.Direction direction = collation.getDirection();
-            DBSPExpression comparator = new DBSPApplyExpression(
-                    DBSPTypeAny.instance.path(
-                            new DBSPPath("Extract", "new")),
-                    row.field(field).closure(row.asRefParameter()));
+            boolean ascending;
             switch (direction) {
                 case ASCENDING:
+                    ascending = true;
                     break;
                 case DESCENDING:
-                    comparator = new DBSPApplyMethodExpression("rev", DBSPTypeAny.instance, comparator);
+                    ascending = false;
                     break;
+                default:
```

Comment:
Where do these sort orders come from? An order by? From the sound of it it looks like a column definition?

Reply:
This is in the function visitSort, which implements the OrderBy clause.